astropy/io/fits/header.py

```diff
@@ -449,6 +451,19 @@ def block_iter(nbytes):
             if close_file:
                 fileobj.close()
 
+    @classmethod
+    def _fromcards(cls, cards):
+        header = cls()
+        for idx, card in enumerate(cards):
+            header._cards.append(card)
+            keyword = Card.normalize_keyword(card.keyword)
```

Comment:
This seems to be copied from the `append` method, couldn't this be factored into a separate method?

Reply:
If you can find a good name yes ;)

Reply:
`_add_card`?

Reply:
The method could not contain the `_cards.append` since `.append` can also insert the card at any position ... so it would just take care of the indices.

Reply:
Could you elaborate? I don't understand the previous comment.